app/forms/school_search_form.rb

```diff
@@ -3,23 +3,70 @@
 class SchoolSearchForm
   include ActiveModel::Model
 
-  attr_accessor :school_name, :location, :search_distance, :search_distance_unit, :characteristics, :partnership
+  CHARACTERISTICS = [OpenStruct.new(id: "pupil_premium_above_40", name: "Pupil premium above 40%"),
+                     OpenStruct.new(id: "top_20_remote_areas", name: "Located in remote area")].freeze
+
+  attr_accessor :school_name,
+                :location,
+                :search_distance,
+                :search_distance_unit,
+                :characteristics,
+                :partnership,
+                :local_authorities,
+                :networks,
+                :search_type,
+                :cohort_year,
+                :lead_provider_id,
+                :selected_cohort_id
 
   def find_schools(page)
-    schools = School.where("lower(name) LIKE ? OR
-                            lower(urn) LIKE ?",
-                           "%#{(school_name || '').downcase}%",
-                           "%#{(school_name || '').downcase}%")
-    .includes(:network, :lead_provider)
+    schools = School.where("schools.name ILIKE ? OR schools.urn ILIKE ?", "%#{school_name || ''}%", "%#{school_name || ''}%")
+                    .includes(:network, :lead_providers)
+
+    schools = schools.unpartnered(year) unless show_partnered?
+
+    schools = schools.partnered(year) if show_partnered?
+
+    schools = schools.partnered_with_lead_provider(lead_provider_id) if lead_provider_id
 
-    schools = schools.where(id: Partnership.pluck(:school_id)) if filter_by_partnership_status
+    schools = schools.with_local_authority(local_authorities) if local_authorities&.reject(&:blank?)&.any?
+
+    schools = schools.where(network: networks) if networks&.reject(&:blank?)&.any?
```

Comment:
ditto

Reply:
Will refactor

Reply:
Refactored and removed most of Ethans code